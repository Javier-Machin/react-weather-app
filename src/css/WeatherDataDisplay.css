.weather-values {
  display: flex;
  justify-content: space-around;
  align-items: center;
  font-size: 1.5rem;
  padding: 3rem 0;
  margin-top: -1rem;
  position: relative;
  text-transform: uppercase;
  color: white;
  z-index: 25;
  background-color: rgba(215, 86, 44, 1);
  border-radius: 0;
  border: 1.8px solid rgba(0, 0, 0, 0.78);
  border-width: 1.8px 1.8px 1.8px 1.8px;
  font-family: sans-serif;
  letter-spacing: 0.3rem;
}
