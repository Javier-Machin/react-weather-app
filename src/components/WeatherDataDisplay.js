import React from 'react';
import '../css/WeatherDataDisplay.css';

const WeatherDataDisplay = props => {
  const { statusCode } = props;

  // Return different elements depending on request status code

  switch (statusCode) {
    case 200:
      const {
        temperature,
        skyStatus,
        rainProbability,
        location,
        humidity
      } = props.weatherData;

      return (
        <section className="weather-values">
          <p>{location}</p>
          <p>{temperature}ÂºC</p>
          <p>{skyStatus}</p>
          <p>Humidity: {humidity}%</p>
          <p>{rainProbability}</p>
        </section>
      );
    case 404:
      return <section className="weather-values">City not found</section>;
    case 503:
      return <section className="weather-values">Service unavailable</section>;
    default:
      return null;
  }
};

export default WeatherDataDisplay;
