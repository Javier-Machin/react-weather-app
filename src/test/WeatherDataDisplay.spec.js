import React from 'react';
import { shallow } from 'enzyme';
import WeatherDataDisplay from '../components/WeatherDataDisplay';

describe('WeatherDataDisplay', () => {
  it('should render correctly', () => {
    const component = shallow(<WeatherDataDisplay />);
    expect(component).toMatchSnapshot();
  });

  it('should display data correctly', () => {
    const weatherData = {
      temperature: 33,
      skyStatus: 'clear',
      rainProbability: 'unlikely',
      humidity: 33
    };

    const component = shallow(
      <WeatherDataDisplay weatherData={weatherData} statusCode={200} />
    );

    expect(component.text()).toContain('33ÂºC');
    expect(component.text()).toContain('clear');
    expect(component.text()).toContain('unlikely');
    expect(component.text()).toContain('Humidity: 33%');
  });

  it('should display city not found on 404', () => {
    const component = shallow(<WeatherDataDisplay statusCode={404} />);

    expect(component.text()).toContain('City not found');
  });

  it('should display service unavailable on 503', () => {
    const component = shallow(<WeatherDataDisplay statusCode={503} />);

    expect(component.text()).toContain('Service unavailable');
  });
});
