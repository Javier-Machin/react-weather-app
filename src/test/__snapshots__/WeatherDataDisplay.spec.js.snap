// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDataDisplay should render correctly 1`] = `""`;
